### PR TITLE
Allow a range of package:web dependencies

### PR DESCRIPTION
**Pull request by ditman**
Revert "bump to web:^1.0.0"

This reverts commit 696b254341bc75a407d827c2835f55d961301959.

### CODE REVIEW

**Comment by ditman** [comment]
## But why?

This would help us smoothly land the following PRs in order:

* https://github.com/flutter/packages/pull/7077 (migration to google_maps 8.0.0), and then
* https://github.com/flutter/packages/pull/7202 (migration to web 1.0.0)

I had to do something similar with my js-interop package for GIS, see here:

* https://github.com/flutter/packages/pull/7203

---

@a14n is there any problem that I'm not seeing by maintaining the >=0.5.1 <2.0.0 range? It seems I didn't need any code changes in the JS-interop code itself, only in the examples (that are assuming 1.0.0 already)

**Comment by a14n** [comment]
If we go back to  `web: ">=0.5.1 <2.0.0"` and `dart pub downgrade` then there are analysis errors on `innerHTML` assignments in examples like:
>A value of type 'JSString' can't be assigned to a variable of type 'String'.
Try changing the type of the variable, or casting the right-hand type to 'String'.

That's why I changed that. But if it makes your migration easier I can publish a temporary version. I just don't want to have this kind of issue with the final release.

**Comment by a14n** [comment]
8.0.0-dev.4 published

**Comment by ditman** [comment]
> there are analysis errors on `innerHTML` assignments in examples

But `example` is not part of the `lib` that people use, right?

Why not make the `example` an actual app, so it has its own `pubspec.yaml`? That way we can set the dependency to `web: ^1.0.0` there (that's what I did with the GIS web sdk at least).

I haven't looked deep at how examples work/are built in this package, but I can set `example` as a separate app, if it helps!

**Comment by ditman** [comment]
Another way of supporting the old innerHTML behavior in `web:>=0.5.1 <2.0.0` is with something like this :)

https://github.com/flutter/packages/pull/7077/commits/560bda454bb20edeef351a3dcf3f6cada7ba8956#diff-cf680b79805a49a4b4b228b2bc0b5060c31ca4f19aa801af996bf120ebd20575R36-R39

**Comment by Rexios80** [comment]
I agree with examples being their own app if they cause issues

**Comment by a14n** [comment]
Forget my concern about examples. It's fine. Let me know if I can publish a final version of 8.0.0.

File: pubspec.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ environment:
   sdk: ^3.4.0
```
Comment (ditman) [inline pubspec.yaml:8]: I just realized a small mistake in v8.0.0; in order for the package:web version bracket to work, this `sdk` dependency should be: `sdk: ^3.3.0`.

Requiring `^3.4.0` here is (almost) the same as doing `web: ^1.0.0`.

I'll try to create a mini PR ASAP. Sorry for the churn, the new version is working great (and in Wasm!)

**Pull request merged**